tests/rados/test_scrub_omap.py

```diff
@@ -0,0 +1,119 @@
+"""
+This file contains the  methods to verify the  OSD status during large omap entries.
+AS part of verification the script  perform the following tasks-
+   1. Creating omaps
+   2. Perform scrub and deepscrub
+   3. While creating objects check that osds status
+"""
+
+
+import datetime
+import time
+from threading import Thread
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.pool_workflows import PoolFunctions
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Test to create a large number of omap entries on the single PG pool and test osd resiliency
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    pool_obj = PoolFunctions(node=cephadm)
+
+    pool_target_configs = config["verify_osd_omap_entries"]["configurations"]
+    omap_target_configs = config["verify_osd_omap_entries"]["omap_config"]
+
+    # Creating pools and starting the test
+    for entry in pool_target_configs.values():
+        log.debug(
+            f"Creating {entry['pool_type']} pool on the cluster with name {entry['pool_name']}"
+        )
+        if entry.get("pool_type", "replicated") == "erasure":
+            method_should_succeed(
+                rados_obj.create_erasure_pool, name=entry["pool_name"], **entry
+            )
+        else:
+            method_should_succeed(
+                rados_obj.create_pool,
+                **entry,
+            )
+
+        log.debug(
+            "Created the pool. beginning to create large number of omap entries on the pool"
+        )
+    omap_thread = Thread(
+        target=create_omap, args=[pool_obj, pool_target_configs, omap_target_configs]
+    )
+
+    # verification of omap for 30 minutes
+    verification_osd_thread = Thread(target=verification_osd, args=[rados_obj, 30])
+    omap_thread.daemon = True
+    omap_thread.start()
+    verification_osd_thread.start()
+    verification_osd_thread.join()
```

Comment:
We are starting the Verification tests even before we are done with writing the required no of omaps on the pool. Verification needs to start post omaps are written

Reply:
The verification will happen in parallel with the omap creations. Both threads are running for 30 minutes, In these 30 minutes one thread is creating omaps and another is verifying the OSD failures. For every three seconds, the OSD check will happen. If we want to make a delay to verify the OSD failure I will add in the verification_osd_thread. 

Log snippet:-
-------------------
2023-03-16 07:10:36,313 (cephci.test_scrub_omap) [INFO] - cephci.tests.rados.test_scrub_omap.py:112 - The total number of osd are:15
2023-03-16 07:10:36,314 (cephci.test_scrub_omap) [INFO] - cephci.tests.rados.test_scrub_omap.py:114 - The total number of osd up are:15
2023-03-16 07:10:36,391 (cephci.test_scrub_omap) [DEBUG] - cephci.ceph.ceph.py:1507 - b'wrote 4200000 of 4000000000 omap entries'
2023-03-16 07:10:36,892 (cephci.test_scrub_omap) [DEBUG] - cephci.ceph.ceph.py:1507 - b'wrote 4300000 of 4000000000 omap entries'
2023-03-16 07:10:37,382 (cephci.test_scrub_omap) [DEBUG] - cephci.ceph.ceph.py:1507 - b'wrote 4400000 of 4000000000 omap entries'
2023-03-16 07:10:37,879 (cephci.test_scrub_omap) [DEBUG] - cephci.ceph.ceph.py:1507 - b'wrote 4500000 of 4000000000 omap entries'
2023-03-16 07:10:38,394 (cephci.test_scrub_omap) [DEBUG] - cephci.ceph.ceph.py:1507 - b'wrote 4600000 of 4000000000 omap entries'
2023-03-16 07:10:38,948 (cephci.test_scrub_omap) [DEBUG] - cephci.ceph.ceph.py:1507 - b'wrote 4700000 of 4000000000 omap entries'
2023-03-16 07:10:39,316 (cephci.test_scrub_omap) [INFO] - cephci.tests.rados.test_scrub_omap.py:112 - The total number of osd are:15
2023-03-16 07:10:39,317 (cephci.test_scrub_omap) [INFO] - cephci.tests.rados.test_scrub_omap.py:114 - The total number of osd up are:15
2023-03-16 07:10:39,486 (cephci.test_scrub_omap) [DEBUG] - cephci.ceph.ceph.py:1507 - b'wrote 4800000 of 4000000000 omap entries'
2023-03-16 07:10:40,012 (cephci.test_scrub_omap) [DEBUG] - cephci.ceph.ceph.py:1507 - b'wrote 4900000 of 4000000000 omap entries'
2023-03-16 07:10:40,598 (cephci.test_scrub_omap) [DEBUG] - cephci.ceph.ceph.py:1507 - b'wrote 5000000 of 4000000000 omap entries'

Reply:
Excatly my point. Even by the end of execution, We just have ` 260300000/4000000000 `, which is just around 6% of objects and omaps palnned to be written.

The issue was seen when large number of omaps are present on pools and scrub/deep-scrubs are initiated.
for this, We need to have large number of OMAPs in pool already.

``` 2023-03-16 07:39:37,942 (cephci.test_scrub_omap) [INFO] - cephci.ceph.ceph.py:1543 - Running command podman --version | awk {'print $3'} on 10.0.208.184 timeout 600
2023-03-16 07:39:37,995 (cephci.test_scrub_omap) [INFO] - cephci.run.py:1053 - Podman Version 4.2.0
2023-03-16 07:39:37,996 (cephci.test_scrub_omap) [INFO] - cephci.ceph.ceph.py:1543 - Running command docker --version | awk {'print $3'} on 10.0.208.184 timeout 600
2023-03-16 07:39:38,014 (cephci.test_scrub_omap) [DEBUG] - cephci.ceph.ceph.py:1507 - b'wrote 260300000 of 4000000000 omap entries'
2023-03-16 07:39:38,053 (cephci.test_scrub_omap) [INFO] - cephci.ceph.ceph.py:1543 - Running command ceph --version | awk '{print $3}' on 10.0.209.170 timeout 600
2023-03-16 07:39:38,189 (cephci.test_scrub_omap) [INFO] - cephci.run.py:1069 - ceph Version 17.2.5-75.el9cp
2023-03-16 07:39:38,192 (cephci.test_scrub_omap) [INFO] - cephci.run.py:1028 - ceph_clusters_file rerun/rados_omap-GTZEK4 ```

Reply:
Creating large omaps and code included to check for large omap created or not.